otherlibs/systhreads/st_stubs.c

```diff
@@ -423,7 +442,9 @@ CAMLprim value caml_thread_initialize(value unit)   /* ML */
   /* Set up a thread info block for the current thread */
   curr_thread =
     (caml_thread_t) caml_stat_alloc(sizeof(struct caml_thread_struct));
-  curr_thread->descr = caml_thread_new_descriptor(Val_unit);
+  curr_thread->uniq_ident = caml_thread_new_ident();
+  ST_CHANGE_HOOK(curr_thread->uniq_ident,ST_HOOK_INIT);
```

Comment:
Calling the hook at this point means it cannot go through `all_threads` and recover the thread's descriptor. Wouldn't it be better to call the hook after all the fields are initialized and the record is inserted in the list?